funding/manager.go

```diff
@@ -2193,10 +2274,41 @@ func makeFundingScript(channel *channeldb.OpenChannel) ([]byte, error) {
 // NOTE: This MUST be run as a goroutine.
 func (f *Manager) waitForFundingConfirmation(
 	completeChan *channeldb.OpenChannel, cancelChan <-chan struct{},
-	confChan chan<- *confirmedChannel) {
+	confChan chan<- *confirmedChannel, errChan chan error) {
 
 	defer f.wg.Done()
 	defer close(confChan)
+	// First we wait for any spend of the inputs to the funding
+	// transaction. We do this to handle the case where any of the
+	// inputs gets spent by another transaction, invalidating this
+	// funding tx.
+	// TODO(halseth): Move this logic to within RegisterConfNtfn
+	// and return error in case of double spend?
+	if completeChan.IsInitiator && completeChan.ChanType.HasFundingTx() {
+
+		fundingTxid := completeChan.FundingTxn.TxHash()
+		log.Infof("Waiting for funding tx (%v) spend event",
+			fundingTxid)
+
+		spendTxid, err := f.waitForFundingSpend(completeChan.FundingTxn,
+			completeChan.FundingBroadcastHeight)
+		if err != nil {
+			log.Errorf("Failed waiting for funding spend: %v",
+				err)
+			return
+		}
+
+		// If the funding transaction is being invalidated by a tx
+		// spending one of its inputs, we can cancel this funding flow.
+		if *spendTxid != fundingTxid {
+			log.Warnf("Input to funding tx spent by another "+
+				"tx(%v)", spendTxid)
+
+			errChan <- ErrDoubleSpend
```

Comment:
We should also clear the internal funding manager state (ideally add another state that indicates things are being cancelled, so we can execute the transition reliably on restart), and then also clear out the active/pending channel state in the `channeldb`.

Reply:
Should this not be happening here? https://github.com/johng/lnd/blob/1e63f258bd434451dd902c0147262ae2c1fa5a4d/funding/manager.go#L1128 It seems like here is common code for cleaning up pending channels